near-sdk/src/store/tree_map/mod.rs

```diff
@@ -0,0 +1,2118 @@
+    {
+        self.remove_entry(key).map(|(_, v)| v)
+    }
+}
+
+enum Edge {
+    Left,
+    Right,
+}
+
+impl<K> Tree<K>
+where
+    K: Ord + BorshSerialize + BorshDeserialize,
+{
+    fn node(&self, id: FreeListIndex) -> Option<&Node<K>> {
+        self.nodes.get(id)
+    }
+
+    /// Returns the smallest stored key from the tree
+    fn min(&self) -> Option<&K> {
+        let root = self.root?;
+        self.min_at(root).map(|((_, n), _)| &n.key)
+    }
+
+    /// Returns the largest stored key from the tree
+    fn max(&self) -> Option<&K> {
+        let root = self.root?;
+        self.max_at(root).map(|((_, n), _)| &n.key)
+    }
+
+    /// Returns the smallest key that is strictly greater than key given as the parameter
+    fn higher<Q>(&self, key: &Q) -> Option<&K>
+    where
+        K: Borrow<Q>,
+        Q: ?Sized + Ord,
+    {
+        let root = self.root?;
+        self.above_at(root, key)
+    }
+
+    /// Returns the largest key that is strictly less than key given as the parameter
+    fn lower<Q>(&self, key: &Q) -> Option<&K>
+    where
+        K: Borrow<Q>,
+        Q: ?Sized + Ord,
+    {
+        let root = self.root?;
+        self.below_at(root, key)
+    }
+
+    fn equal_key<Q>(&self, key: &Q) -> Option<&K>
+    where
+        K: Borrow<Q>,
+        Q: ?Sized + Ord,
+    {
+        self.root.map(|root| self.equal_at(root, key)).unwrap_or_default()
+    }
+
+    fn floor_key<Q>(&self, key: &Q) -> Option<&K>
+    where
+        K: Borrow<Q>,
+        Q: ?Sized + Ord,
+    {
+        if let Some(key) = self.equal_key(key) {
+            Some(key)
+        } else {
+            self.lower(key)
+        }
+    }
+
+    fn ceil_key<Q>(&self, key: &Q) -> Option<&K>
+    where
+        K: Borrow<Q>,
+        Q: ?Sized + Ord,
+    {
+        if let Some(key) = self.equal_key(key) {
+            Some(key)
+        } else {
+            self.higher(key)
+        }
+    }
+
+    /// Returns (node, parent node) of left-most lower (min) node starting from given node `at`.
+    /// As min_at only traverses the tree down, if a node `at` is the minimum node in a subtree,
+    /// its parent must be explicitly provided in advance.
+    fn min_at(&self, mut at: FreeListIndex) -> Option<(NodeAndIndex<K>, Option<NodeAndIndex<K>>)> {
+        let mut parent: Option<NodeAndIndex<K>> = None;
+        loop {
+            let node = self.node(at);
+            match node.and_then(|n| n.lft) {
+                Some(lft) => {
+                    parent = Some((at, expect(node)));
+                    at = lft;
+                }
+                None => {
+                    return node.map(|node| ((at, node), parent));
+                }
+            }
+        }
+    }
+
+    /// Returns (node, parent node) of right-most lower (max) node starting from given node `at`.
+    /// As min_at only traverses the tree down, if a node `at` is the minimum node in a subtree,
+    /// its parent must be explicitly provided in advance.
```

Comment:
> if a node `at` is the minimum node in a subtree, its parent must be explicitly provided in advance.

I think this bit is outdated, you no longer accept a parent node in the function signature